app/web/src/features/search/SearchBox.test.tsx

```diff
@@ -31,34 +33,54 @@ const View = ({
 };
 
 describe("SearchBox", () => {
-  it("performs a search with the enter key", async () => {
+  it("performs a keyword search", async () => {
     const setActive = jest.fn();
     render(<View setActive={setActive} />, { wrapper });
-    const input = screen.getByLabelText(USER_SEARCH);
-    userEvent.type(input, "test search{enter}");
+    userEvent.click(screen.getByLabelText(SEARCH_BY_KEYWORD));
+    const input = screen.getByLabelText(PROFILE_KEYWORDS);
+    userEvent.type(input, "test search");
     await waitFor(() => {
       expect(setActive).toBeCalledWith({ query: "test search" });
     });
   });
 
-  it("performs a search with the button", async () => {
+  it("performs a location search", async () => {
+    server.listen();
     const setActive = jest.fn();
     render(<View setActive={setActive} />, { wrapper });
-    const input = screen.getByLabelText(USER_SEARCH);
-    userEvent.type(input, "test search");
-    userEvent.click(screen.getByRole("button", { name: SEARCH }));
+    const input = screen.getByLabelText(LOCATION);
+    userEvent.type(input, "tes{enter}");
+    userEvent.click(await screen.findByText("test city, test country"));
     await waitFor(() => {
-      expect(setActive).toBeCalledWith({ query: "test search" });
+      expect(setActive).toBeCalledWith({
+        location: "test city, test country",
+        lng: 1.0,
+        lat: 2.0,
+      });
     });
+    server.close();
```

Comment:
I think it's better to put this in an `afterEach` so it closes the server even if this test fails, so it doesn't affect other tests.